lib/libc/Kconfig

```diff
@@ -80,6 +80,16 @@ endif # NEWLIB_LIBC
 
 if MINIMAL_LIBC
 
+config MINIMAL_LIBC_MALLOC
+	bool "Enable minimal libc malloc implementation"
+	default y
+	help
+	  Enable the minimal libc's implementation of malloc, free, calloc,
+	  realloc, and reallocarray. Disable if you wish to provide your own
+	  implementations of these functions.
```

Comment:
FWIW: you could consider leaving calloc() and reallocarray() in place (or make it configurable), as they're trivial wrappers around malloc/realloc, which is likely going to be true of all user implementations too.

Reply:
I can imagine an non-trivial implementation of calloc, so making it configurable sounds reasonable to me. I'll do the same for reallocarray for consistency, even though I can't think of a way to make a useful non-trivial implementation of that.